### ISSUE TITLE
Hide smart app banner in web app

### ISSUE THREAD

**Issue by ahmedshalaby2018** [ain/smartbanner.js#482]
I need to hide smart app banner in web  app I used this tags
"smartbanner:exclude-user-agent-regex" content="^.Win$"
 but not working , how can I hide it ?

Thanks

![لقطة الشاشة 2022-05-06 060045](https://user-images.githubusercontent.com/38386983/167059871-5521f7d3-c831-4ec7-943d-01e876fe4c93.png)

**Comment by ahmedshalaby2018** [comment]
thanks it's works fine with .*Windows NT.* 
but Can I exclude all website frame work mac windows etc .... ?

**Comment by ain** [comment]
@ahmedshalaby2018 `smartbanner.js` is meant to be used on mobile. Promoting the app on desktop is usually not desired, it should not appear on desktop.

Nevertheless, you can indeed extend the exclusion regex by piping the platforms, e.g. `.*(Windows NT|Intel Mac OS X).*` etc.

**Comment by citrusella** [comment]
This might not be related to why the original creator of this issue was asking this, but for what it's worth, a website I use that I'm pretty sure is using this to promote their app on mobile recently updated from 1.17 to 1.19 (with no other changes I can find to the banner code or anything) and I pretty much immediately started seeing iOS app promotion on my Windows 10 laptop computer. All I can ascertain is that this is because between those two versions, catching iPad 13+ appears to have been added by checking if the browser has more than 0 maxTouchPoints (and because my computer is a touch screen, my browser tells it that it *does* in fact have more than 0 touch points and so I see the banner). I may be sniffing up the wrong tree but I saw this issue and felt like it was worth mentioning.

(And as a note, I've reported this to the website itself so they know about it; despite my report there being recent enough that I haven't heard back (and normally I would be very patient and wait), I'm mentioning it here too because I found this while trying to investigate the cause myself and didn't want to forget I found it.)

TL;DR: I think in trying to cover iOS devices with user agents that aren't clearly mobile user agents, it also resulted in unintended overlap with desktop/laptop devices that definitely aren't mobile but which happen to have touch points, so some desktop users (namely, ones with touchscreens) are seeing iOS banners.
